### DISCUSSION TITLE
elastic-endpoint consuming high cpu resources when elk down

### DISCUSSION THREAD

**Discussion by lxmacc** [Security-Onion-Solutions/securityonion#11988] [2.4]
### Version

2.4.30

### Installation Method

Security Onion ISO image

### Description

other (please provide detail below)

### Installation Type

Standalone

### Location

airgap

### Hardware Specs

Meets minimum requirements

### CPU

8

### RAM

32GB

### Storage for /

65GB

### Storage for /nsm

650GB

### Network Traffic Collection

tap

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

Hi, 
I've noticed that the endpoint process is consuming alot of cpu (+100%) when the grid node is down.
Are others facing the same issue, if so, is this a common behaviour or a bug?

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by defensivedepth** [comment]
A few questions:
- What OS is the endpoint?
- What version of Elastic Agent is it running (there was an Elastic upgrade in 2.4.30, and deployed agents arent automatically upgraded)
- Are you using any additional Integrations other than the defaults?

**Reply by lxmacc** [reply]
The endpoints are running on RHEL8.
Elastic Agent version is 8.10.4

Used Integrations are;
Elastic Defend
Osquery Manager
System
Windows

By removing the Elastic Defend integration from the agent policy the issue is solved. So i suppose the problem comes from the integration itself.
Malware Detection is turned off. Also tried to tweak logs gathering by disabling process, files or network but the issue still persists.

**Reply by defensivedepth** [reply]
I'm going to attempt to reproduce it - so when the Manager is not accessible, the agent's CPU spikes? For how long?

**Reply by lxmacc** [reply]
right, until the manager is available again

**Reply by defensivedepth** [reply]
Ok, please allow us some time to dig into this internally.

**Reply by lxmacc** [reply]
I did a test on a clean elk 11.8.2 install and was not able to reproduce the issue

**Reply by dougburks** [reply]
> I did a test on a clean elk 11.8.2 install and was not able to reproduce the issue

Do you mean Elastic version 8.11.2? We're currently on Elastic 8.10.4, so it's possible this is an Elastic bug that was resolved between 8.10.4 and 8.11.2.

**Reply by lxmacc** [reply]
Yes, I am aware that SOC runs Elastic version 8.10.4. 
Did you already had the chance, trying to reproduce the error on your side?

**Reply by lxmacc** [reply]
I have a distributaed installation in mean time with a manager and receiver node. The issue persists when the manager node is down.
It seems that the issue comes from the Elastic Defend integration. The problem does not occure when log collection in Elastic Defend is disabled.

**Reply by defensivedepth** [reply]
@lxmacc Have you created an issue over at Elastic?

**Reply by lxmacc** [reply]
No, as I am not able reproduce the issue on a standalone ELK. 
I've just tested on a 8.10.4 install of ELK and the issue does only occure on the Security Onion install.

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/153628259/e2bc1cbd-b219-43a4-92fb-127b0395e3e4)

I've install the elk agent around 12:00 and shutdown the cluster until 14:00.
Around 14:00 I' ve installed the so-agent and turned of the so server around 14:30.

**Reply by hoalex** [reply]
@defensivedepth Did you have time to investigate this problem? I have the same issue, and it is somewhat of a blocker. Imagine the manager down going down (for whatever reason), and suddenly the CPU begins to spike on all VMs that have Elastic Agent running.
From what I can see in the documentation, there is no full redundancy available for the manager role, except for the receiver node?

**Reply by defensivedepth** [reply]
@lxmacc @hoalex When you attempted to recreate with a stock ELK stack, did you use the Elastic Defend integration? I believe that is the issue. As a workaround, you can just remove that integration from the default agent policy.
